client/src/components/context/receipt/ReceiptState.js

```diff
@@ -87,18 +47,142 @@ const ReceiptState = (props) => {
   };
 
   // Get receipts
+  const getReceiptsByMonth = async (obj) => {
+    console.log('get By year and month fired');
+    const config = {
+      params: {
+        month: obj.month,
+        year: obj.year,
+      },
+      header: {
+        'Content-Type': 'application/json',
+      },
+    };
+
+    try {
+      const res = await axios.get('/getReceipts', config);
+      console.log(res);
+      const data = {
+        response: await res.data.response.reverse(),
+        status: await res.data.status,
+        total_amount: await res.data.total_amount,
+      };
+
+      dispatch({
+        type: GET_RECEIPTS,
+        payload: data,
+      });
+    } catch (error) {
+      dispatch({
+        type: RECEIPT_ERROR,
+        payload: 'Something went wrong',
+      });
+    }
+  };
+
+  // Get All Receipts
+  const getAllReceipts = async () => {
+    console.log('get all fired');
+    const config = {
+      header: {
+        'Content-Type': 'application/json',
+      },
+    };
+
+    try {
+      const res = await axios.get('/viewAllReceipts', config);
+      console.log(res);
+      const data = {
+        response: await res.data.response.reverse(),
+        status: await res.data.status,
+        total_amount: await res.data.total_amount,
+      };
+
+      dispatch({
+        type: GET_ALL_RECEIPTS,
+        payload: data,
+      });
+    } catch (error) {
+      dispatch({
+        type: RECEIPT_ERROR,
+        payload: 'Something went wrong',
+      });
+    }
+  };
+
+  // Get Receipts by Year
+
+  const getReceiptsByYear = async (year) => {
+    console.log('get By year fired');
+
+    const config = {
+      header: {
+        'Content-Type': 'application/json',
+      },
+    };
+
+    try {
+      const res = await axios.get('/viewAllReceipts', config);
+      console.log(res);
+      const items = await res.data.response;
+      const filteredItems = items
+        .reverse()
```

Comment:
i think you use `reverse` to display the latest receipts first? 

perhaps we need to change this in the backend. by default - it should sort by latest created date rather than handling this on the frontend

Reply:
I did reverse because I was just following the response that I was getting from the backend. Thanks for the tip. We will sort this from the backend.